Library/Homebrew/version.rb

```diff
@@ -379,3 +392,56 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
   end
 end
+
+class HeadVersion < Version
+  attr_reader :version
+  attr_reader :commit
+
+  def initialize(val)
+    if val.respond_to?(:to_str)
+      @version = if (val.to_str.start_with?("HEAD"))
+        val.to_str
+      else
+        "HEAD-#{val}"
+      end
+      @commit = @version[/HEAD-(.+)/, 1]
+    else
+      raise TypeError, "Version value must be a string; got a #{val.class} (#{val})"
+    end
+  end
+
+  def update_commit(commit)
+    @commit = commit
+    update_version
+    version
+  end
+
+  def to_s
+    if commit
+      "HEAD-#{commit}"
+    else
+      "HEAD"
+    end
```

Comment:
Actually, you don't need to implement `to_s`. It's already there in parent class.